### ISSUE TITLE
DigraphCore can give wrong results

### ISSUE THREAD

**Issue by wilfwilson** [digraphs/Digraphs#436]
From the documentation:
> The  core  of  a  digraph  D  is  the  minimal  subdigraph C of D which is a homomorphic image of D. The core of a digraph is unique up to isomorphism.

Let's create the following digraph and compute its core.
```
gap> D := DigraphFromDigraph6String("&GSY??A?SA?O?");
<immutable digraph with 8 vertices, 10 edges>
gap> DigraphCore(D);
[ 1, 2, 3 ]
gap> core := InducedSubdigraph(D, DigraphCore(D));
<immutable digraph with 3 vertices, 3 edges>
```
Alright,let's check that there is actually a homomorphism from `D` to `core`:
```
gap> DigraphHomomorphism(D, core) <> fail;
true
```
Good. But, it's not minimal...
```
gap> smaller_core := InducedSubdigraph(D, [1 ,2]);
<immutable digraph with 2 vertices, 2 edges>
gap> IsCompleteDigraph(smaller_core);
true
gap> DigraphHomomorphism(D, smaller_core) <> fail;
true
```
Indeed, `IsDigraphCore` tells us that the core isn't actually a core, directly:
```
gap> IsDigraphCore(core);
false
```

So unless I'm misunderstanding something, the result of `DigraphCore` for `D` is wrong; in particular, it should be `[1, 2]`.

**Comment by wilfwilson** [comment]
I'm pretty sure that, in general:
```
DigraphCore(D) = ImageSetOfTransformation(RepresentativeOfMinimalIdeal(EndomorphismMonoid(D)));
```
(Not that this is necessarily the best way to check it, but it should give us a good way of writing test cases/working out whether this bug is fixed.)

**Comment by wilfwilson** [comment]
Here's a much simpler example where it goes wrong:
```
gap> D := Digraph([[2], [1, 3], []]);;
gap> DigraphCore(D);
[ 1, 2, 3 ]
gap> DigraphHomomorphism(D, InducedSubdigraph(D, [1, 2]));
Transformation( [ 2, 1, 2 ] )
gap> IsDigraphCore(D);
false
```

**Comment by wilfwilson** [comment]
Resolved in v1.4.1.
